### ISSUE TITLE
[Meta] Xcode 14.0 Support

### ISSUE THREAD

**Issue by dalexsoto** [xamarin/xamarin-macios#15954]
# 🚨 Update ⚠️

Hello All!

We have what we believe is the final build for Xcode 14.0 Support
* [xamarin.ios-16.0.0.72.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/6.0.4xx-xcode14/6756a11462d8aa6e73d4a320712b276caba159c1/6718459/package/notarized/xamarin.ios-16.0.0.72.pkg)

If you are currently using **.NET MAUI** you will need to wait a little longer in order to use Xcode 14, as for now we highly encourage you to download Xcode 13.4.1 from the [downloads](https://developer.apple.com/download/all/) page in the Apple Developer Portal and continue to use it for now.

Thank you for your patience!

The macios team!

-----------

<details>
  <summary>Update 0</summary>

~Apple released Xcode 14.0 on September 12, 2022 we are working to get our Xcode 14 support release out in the meantime if you use Xamarin **we recommend you to not upgrade to Xcode 14 just yet** and keep using Xcode 13.4.1.~

~If you updated to Xcode 14, you have 2 options:~
~1. You can download Xcode 13.4.1 from the [downloads](https://developer.apple.com/download/all/) page in the Apple Developer Portal and continue to use it.~
~2. Use a build from our [xcode14](https://github.com/xamarin/xamarin-macios/tree/xcode14) branch, this is where our next stable release is coming out and it includes all the iOS 15 SDKs APIs, that said we are still ironing it out.~
    * ~[xamarin.ios-15.99.0.464.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/xcode14/9a839d6c4a4c31a2b524fcdb721457900b396b84/6679621/package/notarized/xamarin.ios-15.99.0.464.pkg)~

~We will keep this issue updated as we go to have the final builds declared.~

~If you are currently using **.NET MAUI** you will need to wait a little longer in order to use Xcode 14, as for now we highly encourage you to use option 1 above.~

~Thank you for your patience!~

~The macios team!~

</details>

**Comment by Adam-Langley** [comment]
Hi @dalexsoto 
Can you please answer the original question in the original thread #15946 ?

"Where is the definitive source of truth, a page that is kept up to date, indicating compatibility with Xcode?"

Let's resolve the root issue which is the lack of documentation surrounding how to know whether it is safe to do _any_ Xcode upgrade - i.e, a compatibility matrix between the Xamarin, and Xcode releases.

Thanks for your time

**Comment by dalexsoto** [comment]
> Hi @dalexsoto Can you please answer the original question in the original thread #15946 ?
> 
> "Where is the definitive source of truth, a page that is kept up to date, indicating compatibility with Xcode?"
> 
> Let's resolve the root issue which is the lack of documentation surrounding how to know whether it is safe to do _any_ Xcode upgrade - i.e, a compatibility matrix between the Xamarin, and Xcode releases.
> 
> Thanks for your time

As of today for Xamarin packages the major version part matches the iOS version they are meant for, so for example xamarin.ios 16.x.x will be meant to be used with the iOS 16 SDK which is part of Xcode 14. But the true rule of thumb is that you will always want to use the latest Xcode release with the latest stable Xamarin provided by your Visual Studio. The only time it does not entirely apply is when Xcode has major releases where we are always a little behind because we get Xcode the same day as everyone else so we need some time to catch up and produce builds with the latest release.

Thanks!

**Comment by grvingit** [comment]
Thank you for this information, I am really grateful to you that I am up and running again. I installed XCode 14 and the problems started. I have uninstalled 14 and installed 13.4.1 and it's all working fine. Touch wood.
I second what Adam-Langley said, this should be clearly documented somewhere, not hidden in a single bug post here.
Just a clear list of current recommended versions of all frameworks.

**Comment by filipkristo** [comment]
With option two I have error "failed to compile the generated registrar code" when I want to deploy to device with development provisioning profile. Anyone else have similar error?

**Comment by idzikovski** [comment]
> With option two I have error "failed to compile the generated registrar code" when I want to deploy to device with development provisioning profile. Anyone else have similar error?

Yes, I do.

**Comment by filipkristo** [comment]
> > With option two I have error "failed to compile the generated registrar code" when I want to deploy to device with development provisioning profile. Anyone else have similar error?
> 
> Yes, I do. But I guess it makes sense if your phone is still on iOS 15 as mine. I will update to iOS 16 and try it again

Mine is on iOS 16

**Comment by emanuelecastelli** [comment]
Here: VS for mac 17.3.5

**Comment by emanuelecastelli** [comment]
VS for Mac 17.3.5 with xamarin.ios-15.99.0.464
Xcode 14
iOS 16

I get following build error:
`MTOUCH: Error MT4109: Failed to compile the generated registrar code. Please file a bug report at https://github.com/xamarin/xamarin-macios/issues/new (MT4109)`

**Comment by banditoth** [comment]
Happening for me aswell.

**Comment by vanyok1991** [comment]
Have the same issue

**Comment by dmoodz** [comment]
+1

**Comment by timahrentlov** [comment]
+1

**Comment by Cheesebaron** [comment]
The +1 and "have same issue" comments are not valuable, use reactions at the opening post.

**Comment by timahrentlov** [comment]
> The +1 and "have same issue" comments are not valuable, use reactions at the opening post.

+1

**Comment by wispborne** [comment]
The Xamarin quickstart page is not accurate as of XCode 14's release: https://docs.microsoft.com/en-us/xamarin/ios/get-started/hello-ios/hello-ios-quickstart?pivots=macos#requirements

> iOS development with Xamarin requires:
> - Latest version of Xcode and iOS SDK installed from the [App Store](https://itunes.apple.com/us/app/xcode/id497799835?mt=12) .

**Comment by tipa** [comment]
> * [xamarin.ios-15.99.0.464.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/xcode14/9a839d6c4a4c31a2b524fcdb721457900b396b84/6679621/package/notarized/xamarin.ios-15.99.0.464.pkg)

Is this for legacy Xamarin only? I did download & install that preview package but still getting an error when trying to compile my .NET6 (non-MAUI) iOS app.

```
Tool /usr/local/share/dotnet/packs/Microsoft.NETCore.App.Runtime.AOT.osx-x64.Cross.ios-arm64/6.0.9/Sdk/../tools/mono-aot-cross execution finished (exit code = 134).
/usr/local/share/dotnet/packs/Microsoft.iOS.Sdk/15.4.447/targets/Xamarin.Shared.Sdk.targets(1000,3): error : Failed to AOT compile SQLitePCLRaw.provider.sqlite3.dll, the AOT compiler exited with code 134 [/path_to_csproj]
```

**Comment by ivanicin** [comment]
@pm64 while I have notable remarks on Xamarin that I decided to keep silent at the moment, but I think that this was answered. 

Microsoft tries to support the latest version of XCode with the latest version of BS. As XCode may break this without any previous notice there is no practical way to inform people. They just try to issue update as soon as possible. 

That’s from me to save some time so that people can work now when they should work and not answer the questions.

**Comment by protokovich** [comment]
17.4.0 Preview 2.0
problem still exists.

xamarin.ios-15.99.0.464
installed on Mac.

build is ok on simulator. but error on device (iPhone 12 mini iOS 16):

> C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Xamarin\iOS\Xamarin.iOS.Common.targets(235,3): error GE71D2E27: no type or protocol named 'BADownloadManagerDelegate'
> 4>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Xamarin\iOS\Xamarin.iOS.Common.targets(235,3): error MT4109: Failed to compile the generated registrar code. Please file a bug report at https://github.com/xamarin/xamarin-macios/issues/new

**Comment by EduardoReisDev** [comment]
I got these two errors when trying to compile a Xamarin project and an iPhone. 

- XCode 14
- Visual Studio 17.3.5
- Xamarin.iOS 15.99.0.464

![image](https://user-images.githubusercontent.com/52722526/190246289-f184ba39-1277-4b24-9750-14d7e54d6646.png)

**Comment by dalexsoto** [comment]
Ah right, Apple removed the [Background Assets](https://developer.apple.com/documentation/backgroundassets?language=objc) framework from Xcode 14 and that build does not account for that just yet, the registrar errors will be gone in the next update. 

The other way you can workaround this is by setting the linker to [Link SDK assemblies only](https://docs.microsoft.com/en-us/xamarin/ios/deploy-test/linker?tabs=macos#link-sdk-assemblies-only), this should get rid of the registrar errors.

**Comment by EduardoReisDev** [comment]
@dalexsoto, resulted in a new error 😢 

![image](https://user-images.githubusercontent.com/52722526/190261644-34846450-ac08-4c40-b2dc-dab9d0de4f75.png)

![image](https://user-images.githubusercontent.com/52722526/190261574-563d14b0-3096-40cd-8f70-35bfb8e15557.png)

**Comment by EduardoReisDev** [comment]
Emulator with iOS 16 working. ✅ 

Real iPhone with iOS 16 not working. ❌

**Comment by EduardoReisDev** [comment]
@dalexsoto, GREAT!!!! Working on a real iPhone and the emulator, both with iOS 16 ✅

**Comment by EduardoReisDev** [comment]
@dalexsoto, GREAT!!!! Working on a real iPhone and the emulator, both with iOS 16 ✅

**Comment by jmmtrimble** [comment]
@dalexsoto  Working fine for me too!! Thanks for the solution.

**Comment by beeradmoore** [comment]
> 1. You can download Xcode 13.4.1 from the [downloads](https://developer.apple.com/download/all/) page in the Apple Developer Portal and continue to use it.

My 2c on that. This tool isn't from MS (and probably isn't endorsed by them in any way) but for those wanting to easily manage, download, and keep multiple versions of Xcode side by side should check out [Xcodes](https://www.xcodes.app).

Minor version increments of Xcode shouldn't break things, but when it does it is a pain to re-download, extract, etc. So it is super handy to keep them side by side and use this tool to select current active version. Then uninstall when you are happy with stability of the latest version. 

VS4Mac also jumps to the correct version of Xcode when you mark them as active within the app so you don't have to go touch that config either.

**Comment by emanuelecastelli** [comment]
``

> Ah right, Apple removed the [Background Assets](https://developer.apple.com/documentation/backgroundassets?language=objc) framework from Xcode 14 and that build does not account for that just yet, the registrar errors will be gone in the next update.
> 
> The other way you can workaround this is by setting the linker to [Link SDK assemblies only](https://docs.microsoft.com/en-us/xamarin/ios/deploy-test/linker?tabs=macos#link-sdk-assemblies-only), this should get rid of the registrar errors.

Thx for your hint, but here we get same error with or without linker active
```
/mtouch-cache/registrar.h:113:9: fatal error: 'CHIP/CHIP.h' file not found
    #import <CHIP/CHIP.h>
            ^~~~~~~~~~~~~
    1 error generated.
```

**Comment by mobilewares** [comment]
FYI: Just in case anyone (like me) went and installed 13.4.1 directly off the Apple site (alongside 14)  before they knew about the existence of XCodes (also me) - and XCodes isn't detecting you have 13.4.1 installed (yes, also me too) :

If you launch XCode 13.4.1 (or I think from 14? too) - go into the **Preferences** > **Locations** - you can tell it to use the XCode 13.4.1 Command Line tools which is the magic thing needed for VS2022 (on PC or Mac) to launch the 13.4.1 Simulators for debugging.  I think XCodes is doing something like this under the hood too.

![Fchw_RlacAA9dq3](https://user-images.githubusercontent.com/4155775/190339640-31db8d1b-bb54-447c-8f32-78cd9fe68488.jpg)

Note the first time I launched it I had a few popups for security/creation of folders etc, just answered OK to it all and now it's all working smoothly.

XCode14 support would be great though (and I really hope this is being prioritized there) - don't necessarily need specific Xamarin iOS16 SDK support yet - but just want the ability to launch the iPhone14xxx + iOS16 simulators for debugging. I imagine a fair amount of my clients customers went and updated their old devices to iOS16 this week and we really need to support it ASAP.

**Comment by IngweLand** [comment]
I have a different error when trying to build and install on real device.
`Error running 'MyApp.iOS': Failed to install application on device qqq: error HE0004: Could not load the framework 'DVTITunesSoftwareServiceFoundation' (path: /Applications/Xcode.app/Contents/SharedFrameworks/DVTITunesSoftwareServiceFoundation.framework/DVTITunesSoftwareServiceFoundation): dlopen(/Applications/Xcode.app/Contents/SharedFrameworks/DVTITunesSoftwareServiceFoundation.framework/DVTITunesSoftwareServiceFoundation, 0x0001): Library not loaded: '@rpath/ContentDelivery.framework/Versions/A/ContentDelivery' //  Referenced from: '/Applications/Xcode.app/Contents/SharedFrameworks/DVTITunesSoftwareServiceFoundation.framework/Versions/A/DVTITunesSoftwareServiceFoundation' //  Reason: tried: '/usr/lib/swift/ContentDelivery.framework/Versions/A/ContentDelivery' (no such file), '/Library/Frameworks/Xamarin.iOS.framework/Versions/15.12.0.2/lib/mlaunch/mlaunch.app/Contents/MacOS/../Frameworks/ContentDelivery.framework/Versions/A/ContentDelivery' (no such file), '/Applications/Xcode.app/Contents/SharedFram... (4 minutes ago)`

**Comment by itsmuntadhar** [comment]
Having the same issue as @IngweLand 

Also, if you're using Venture you can't use Xcode 13 nor its Command Line Tools

it seems I gotta wait for Xcode 14 support to land

**Comment by berniemills** [comment]
Just going to leave my observation of using Visual Studio for Mac Preview and .Net Maui targeting an iPhone. Was all working fine on xCode 13 then I made the mistake of upgrading to 14. Upgrade took ages and then I was not abel to see the simulators as discussed above. I downloaded 13.4.1 as suggested and renamed original xcode 14 so both could live in the Applications folder.

I then got the simulators back in Visual Studio but when I try and deploy to the iPhone the code downloads then fails to install. 
The app appears to have a Cloud download  icon next to it. 
Visual Studio sits saying something like .. wiating for the debugger to connect to X's iPhone on port 10000 via USB.

If anyone has any ideas then do let me know but I get the feeling Xcode 14 download has modified something that I need to undo.

**Comment by Cheesebaron** [comment]
@berniemills if you start Xcode 13 and go to Preferences > Locations

Then change the Command Line tools back to 13.x.x then you should be good to go again

**Comment by berniemills** [comment]
Thanks @Cheesebaron but I did try that. Rather annoyingly I had .net Maui deploying to iPhone on mac all working fine a day or so ago with Xcode 13 but it's clearly messed up now. Its failing to launch the debugger on the iPhone. Saying it could not locate device support files.  A lot of head scratching ahead I think.

**Comment by itsmuntadhar** [comment]
@berniemills try downloading support files for your phone from here https://github.com/filsv/iOSDeviceSupport

**Comment by ivanicin** [comment]
> The other way you can workaround this is by setting the linker to [Link SDK assemblies only](https://docs.microsoft.com/en-us/xamarin/ios/deploy-test/linker?tabs=macos#link-sdk-assemblies-only), this should get rid of the registrar errors.

@dalexsoto this doesn't appear to work for me. If the app has a Watch extension this won't work. Obviously maybe your solution will work but this workaround cannot be applied successfully...

**Comment by berniemills** [comment]
Many thanks @itsmuntadhar  

> @berniemills try downloading support files for your phone from here https://github.com/filsv/iOSDeviceSupport

Many thanks @itsmuntadhar . This did the trick for me. I had iOS16 support files missing in the Xcode 13.4.1 version and so I downloaded from that link and placed them in the appropriate Xcode.app folder and then I was able to debug the app again.

**Comment by pasha-o** [comment]
Getting the following error now with Xcode 14 - ios 16 - I also replaced the ios-device support bits with ios 16
error HE0004: Could not load the framework 'DVTITunesSoftwareServiceFoundation' (path: /Applications/Xcode.app/Contents/SharedFrameworks/DVTITunesSoftwareServiceFoundation.framework/DVTITunesSoftwareServiceFoundation):		 	0

**Comment by aodegov** [comment]
Xcode version: 14.0 (14A309)
iOS version: 16
Xamarin.Forms version: 16 (20A362)
Visual Studio version: Version 17.3.4
Experiencing 2 following issues from 2 different Xamarin forms apps:
1. `Failed to compile the generated registrar code. Please file a bug report at https://github.com/xamarin/xamarin-macios/issues/new`
2. `error HE0004: Could not load the framework 'DVTITunesSoftwareServiceFoundation' (path: /Applications/Xcode.app/Contents/SharedFrameworks/DVTITunesSoftwareServiceFoundation.framework/DVTITunesSoftwareServiceFoundation): 0`

**Comment by ivanicin** [comment]
As  this is meta bug I guess all related bugs should be linked, I have submitted this one few seconds ago: https://github.com/xamarin/xamarin-macios/issues/15983

**Comment by nissaba** [comment]
> The only time it does not entirely apply is when Xcode has major releases where we are always a little behind because we get Xcode the same day as everyone

as in you have started working on compatibility with Xcode 14 this week when the official version came out, or did you start working on it when the beta came out in June?

**Comment by ivanicin** [comment]
> > The only time it does not entirely apply is when Xcode has major releases where we are always a little behind because we get Xcode the same day as everyone
> 
> as in you have started working on compatibility with Xcode 14 this week when the official version came out, or did you start working on it when the beta came out in June?

I am the developer and I also have some frustration regarding the current situation. As I have suggested above I think that we should now focus on helping those people to ship this as soon as possible and raise the questions like yours after that. Such questions won't bring any immediate benefit but they will destruct people and postpone release for sure.

Also I'll try to provide some helpful information regarding your question.

They did start to work quite a while ago, you can actually track the activity on this site. It is more important question on whether the number of people that worked and time that they dedicated was as usual and as far as I know it wasn't. But as said later on that. It is really not fair to blame people like @chamons that did tremendous work for years that they suddenly became lazy, the problem is obviously somewhere else.

The process is that they need first to make bindings compile and it takes some time. They can just after that fully check Xcode issues and fix them. Clearly Apple adds/changes a lot of things since its release in June so it is not like that they can base their final product on that and have full three months for that.

While most factors for the current state are likely internal, it is also true that Apple has made releases unusually early which further exposed the problems in internal organization...

**Comment by pasha-o** [comment]
I totally agree with you @ivanicin  - you folks work really hard and it's unfair to pass judgment like that. I am very grateful for your work and the team work on this. It's a community and should stay that way. again thank you.

**Comment by nmoschkin** [comment]
Yeah, unfortunately the Mac upgraded XCode without me having any say in the matter.  In fact, it upgraded XCode first, and then asked about upgrading the OS later.  I suppose I will just disable all automatic software updates for the Mac, from here on out (assuming my company hasn't altered a security policy to prevent that).

**Comment by nmoschkin** [comment]
I can confirm that downloading, installing and setting the default tools location to XCode 13.4.1 works.

`[Apple Downloads](https://developer.apple.com/download/all/)` <- Scroll down to find XCode 13.4.1

**Comment by berniemills** [comment]
I found that since I was targeting iOS 16 I then had to download the support tools for 16.0 as there were missing from xcode 13.4.1 where upon  it also worked for me.

**Comment by rsbepvb** [comment]
Deleted Xcode 14, downloaded Xcode 13.4.1, installed and now the simulators are back and VS Mac appears to be working ok again.

**Comment by josejuniorzago** [comment]
I can confirm that downgrade to Xcode 13.4.1 works.

**Comment by tussock** [comment]
I too can confirm that:

- Downloading Xcode 13.4.1 from the link above makes everything work again.
- Xcode hasn't automatically re-upgraded (at least not for the last 3 days)... so everything keeps working.

**Comment by lucapan** [comment]
Same problem here! :(

**Comment by nschoenberg** [comment]
@dalexsoto Thanks for letting us know, really helpful and much appreciated :) 
Can you already share a timeframe with us for iOS 16 support?

Thanks in advance

**Comment by lucapan** [comment]
@itsmuntadhar Yes, installing Xcode 13.4.1 solves the problem. Thanks.

**Comment by jorgeperalta** [comment]
Is there an ETA for this? Downgrading xcode is not an option for me

**Comment by adam-russell** [comment]
Just to add a quick not in case this is useful for others -- I've been using Xcode 14 with the same build of Xamarin.iOS mentioned in the first comment: ([xamarin.ios-15.99.0.464.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/xcode14/9a839d6c4a4c31a2b524fcdb721457900b396b84/6679621/package/notarized/xamarin.ios-15.99.0.464.pkg)

And I've had builds with it in the App Store for about a week, starting shortly after the first comment here was made. I haven't had any strange crashes or user complaints, though that may be partially because everything happens to work fine for my particular app. I wanted to build with Xcode 14 because I also use app extensions, and I have iOS 16 features in them.

I test on devices using TestFlight mostly.

**Comment by houserockr** [comment]
> Just to add a quick not in case this is useful for others -- I've been using Xcode 14 with the same build of Xamarin.iOS mentioned in the first comment: [xamarin.ios-15.99.0.464.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/xcode14/9a839d6c4a4c31a2b524fcdb721457900b396b84/6679621/package/notarized/xamarin.ios-15.99.0.464.pkg)
> 
> And I've had builds with it in the App Store for about a week, starting shortly after the first comment here was made. I haven't had any strange crashes or user complaints, though that may be partially because everything happens to work fine for my particular app. I wanted to build with Xcode 14 because I also use app extensions, and I have iOS 16 features in them.
> 
> I test on devices using TestFlight mostly.

@adam-russell thanks this made my day! I've been fiddling for 2 hours and now I'm back in business.

**Comment by pasha-o** [comment]
@adam-russell just tested your solution with the updated package - my environment is visual studio 22 for windows, the simulators show now and I can compile and run. however, running on the real apple device with ios 16 still shows multiple errors:
1- Failed to compile the generated registrar code. 
2- no type or protocol named 'BADownloadManagerDelegate
3-error HE0004: Could not load the framework 'DVTITunesSoftwareServiceFoundation' (path: /Applications/Xcode.app/Contents/SharedFrameworks/DVTITunesSoftwareServiceFoundation.framework/DVTITunesSoftwareServiceFoundation)

**Comment by Phenek** [comment]
Visual Studio should warn us to not update Xcode.
Regards,

**Comment by Magic73** [comment]
I've downgrade Xcode to 13.4.1, but I'm still getting this error:

This warning:
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Xamarin\iOS\Xamarin.Shared.targets(1730,3): warning : iOS SDK version '15 .4 'is not installed. Alternatively, the latest version '15 .5 'will be used.

and the this error at the end of the build:
Xamarin.iOS 15.12.0 does not support a deployment target of 15.5 for iOS (the maximum is 15.4). Please select an older deployment target in your project's Info.plist or upgrade to a newer version of Xamarin.iOS.			


How may I reinstall iOS SDK 15.4???

**Comment by ivanicin** [comment]
> and the this error at the end of the build:
> Xamarin.iOS 15.12.0 does not support a deployment target of 15.5 for iOS (the maximum is 15.4).

While this won’t answer your question directly (as what you ask is probably possible) do you know what your request means and why would you actually need that (as far as I am aware there is no difference in API from 15.0 so you just insist that users with older devices can’t use your app)?

**Comment by dalexsoto** [comment]
Hello All!

We have what we believe is the final build for Xcode 14.0 Support
* [xamarin.ios-16.0.0.72.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/6.0.4xx-xcode14/6756a11462d8aa6e73d4a320712b276caba159c1/6718459/package/notarized/xamarin.ios-16.0.0.72.pkg)

If you are currently using **.NET MAUI** you will need to wait a little longer in order to use Xcode 14, as for now we highly encourage you to download Xcode 13.4.1 from the [downloads](https://developer.apple.com/download/all/) page in the Apple Developer Portal and continue to use it for now.

Thank you for your patience!

The macios team!

**Comment by dalexsoto** [comment]
Please report back any issues that you encounter! Thank you all again for your patience and for following along in this Xcode journey!

**Comment by Magic73** [comment]
> 

Unfortunately both my mac and my ipad have updated to the latest version. Now I can't build the application anymore. the visual studio version is completely updated, on the mac i reinstalled the xcode version 13.3.1 (sdk 15.4) and 13.4.1 (sdk 15.5).
(Whenever Apple updates something, it's always the same story. a lot of time lost to be able to go back to building.)

with the current configuration, now I get this:
Xamarin.iOS 15.14.0 does not support a deployment target of 15.5 for iOS (the maximum is 15.2). Please select an older deployment target in your project's Info.plist or upgrade to a newer version of Xamarin.iOS.

I'm trying to deploy on my ipad (iOS 15.5).

**Comment by jtorvald** [comment]
@dalexsoto I downloaded and installed the package above and I can confirm that my build works now.

**Comment by TheSundayDev** [comment]
installed last version and it works! thanks a lot!

**Comment by Sivagnanam-molex** [comment]
@dalexsoto  Thanks for this package. It working fine.  Please make it as update for Xamarin.iOS in visual studio.

**Comment by ivanicin** [comment]
@dalexsoto for me it is the same - all bugs remain, but most notably you can't built the app if you have the watchOS extension (already reported: https://github.com/xamarin/xamarin-macios/issues/15954#issuecomment-1248158043).

It says:
```
/Users/ii/Projects/SCl/SCl.Watch/SpeechCentral.Watch.WatchOSApp/obj/iPhone/Debug/actool/cloned-assets/Assets.xcassets: Error: The operation couldn’t be completed. Failed to locate any simulator runtime matching options: {
    BuildVersionString = 20R362;
    Platforms =     (
        "com.apple.platform.watchsimulator"
    );
    VersionString = "9.0";
} (SpeechCentral.Watch.WatchOSApp) actool
```

Just to note that I am deploying to the iPhone so there shouldn't be any watch simulator...

**Comment by IDispose** [comment]
No dice. Tried the update via VS 2022 Mac Updater and also explicitly installing the package using the latest link. Unable to get VS4Mac to display any simulators. Previously had tried updating XCode to 13.4. No errors just get "Lower deployment target to see older iOS simulators"

**Comment by IDispose** [comment]
> Unfortunately both my mac and my ipad have updated to the latest version. Now I can't build the application anymore. the visual studio version is completely updated, on the mac i reinstalled the xcode version 13.3.1 (sdk 15.4) and 13.4.1 (sdk 15.5). (Whenever Apple updates something, it's always the same story. a lot of time lost to be able to go back to building.)
> 
> with the current configuration, now I get this: Xamarin.iOS 15.14.0 does not support a deployment target of 15.5 for iOS (the maximum is 15.2). Please select an older deployment target in your project's Info.plist or upgrade to a newer version of Xamarin.iOS.
> 
> I'm trying to deploy on my ipad (iOS 15.5).

This has been my experience as well. Sometimes days are lost just getting back to where I was and continue. No update so far has been a breeze.

**Comment by Magic73** [comment]
I managed to install Xamarin.IiOS version 15.4.0 on the Mac, but now I have this problem connecting to the Mac:

_The '15 .4.0.0 'version of the Xamarin.iOS SDK installed on the Mac is not compatible with this version of Visual Studio. Automatically install Xamarin.iOS '15 .12.0.2 '? This operation involves overwriting any other Xamarin.iOS installation on the Mac._

and here I am back to where I was after updating Visual Studio, following the Mac update.

So, what should I do? Why the latest version of Visual Studio 2022 is incompatible with any version of XCode / iOS installed and installable on the Mac?

Here my VS info:

Microsoft Visual Studio Enterprise 2022
Version 17.3.4
VisualStudio.17.Release/17.3.4+32901.215
Microsoft .NET Framework
Version 4.8.04084

Xamarin.iOS and Xamarin.Mac SDK   15.12.0.2 (87f98a75e)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.

**Comment by Magic73** [comment]
I have, again, allowed the update to iOS version 15.12, from the Mac link popup.
After the installation, I recompiled, and got this error "again":

error MT0074: Xamarin.iOS 15.12.0 does not support a deployment target of 15.5 for iOS (the maximum is 15.4). Please select an older deployment target in your project's Info.plist or upgrade to a newer version of Xamarin.iOS.

I feel like a hamster in the wheel.

**Comment by rolfbjarne** [comment]
@Magic73 

> error MT0074: Xamarin.iOS 15.12.0 does not support a deployment target of 15.5 for iOS (the maximum is 15.4). Please select an older deployment target in your project's Info.plist or upgrade to a newer version of Xamarin.iOS.

Did you try changing the deployment target in your project?

**Comment by dalexsoto** [comment]
> I managed to install Xamarin.IiOS version 15.4.0 on the Mac, but now I have this problem connecting to the Mac:
> 
> _The '15 .4.0.0 'version of the Xamarin.iOS SDK installed on the Mac is not compatible with this version of Visual Studio. Automatically install Xamarin.iOS '15 .12.0.2 '? This operation involves overwriting any other Xamarin.iOS installation on the Mac._
> 
> and here I am back to where I was after updating Visual Studio, following the Mac update.
> 
> So, what should I do? Why the latest version of Visual Studio 2022 is incompatible with any version of XCode / iOS installed and installable on the Mac?
> 
> Here my VS info:
> 
> Microsoft Visual Studio Enterprise 2022 Version 17.3.4 VisualStudio.17.Release/17.3.4+32901.215 Microsoft .NET Framework Version 4.8.04084
> 
> Xamarin.iOS and Xamarin.Mac SDK 15.12.0.2 ([87f98a7](https://github.com/xamarin/xamarin-macios/commit/87f98a75edaa6757fd6ff5170d297615830fb41b)) Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.

@Magic73 you can try this suggestion [here](https://github.com/xamarin/xamarin-macios/issues/12778#issuecomment-925507681) I am not 100% it will work since the VS Windows Integration is now owned by our team but it is worth the shot.

**Comment by IDispose** [comment]
> > No dice. Tried the update via VS 2022 Mac Updater and also explicitly installing the package using the latest link. Unable to get VS4Mac to display any simulators. Previously had tried updating XCode to 13.4. No errors just get "Lower deployment target to see older iOS simulators"
> 
> Please make sure that you Visual Studio for Mac / Xcode settings are pointing to the right Xcode version. Also make sure that after you change any of these settings and after you install [xamarin.ios-16.0.0.72.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/6.0.4xx-xcode14/6756a11462d8aa6e73d4a320712b276caba159c1/6718459/package/notarized/xamarin.ios-16.0.0.72.pkg) you restart Visual Studio For Mac.
> 
> VS Settings: ![image](https://user-images.githubusercontent.com/204671/191761462-9d153890-fc76-496a-bb98-ad7c648d61bf.png)
> 
> Xcode settings: ![image](https://user-images.githubusercontent.com/204671/191762208-17f212b0-dd5d-4bd5-8fbe-db14e823fa0f.png)

My XCode version matches the screenshot. However the xcode app name is just `xcode.app` and not `xcode_14.0.0.app`

**Comment by dalexsoto** [comment]
> My XCode version matches the screenshot. However the xcode app name is just `xcode.app` and not `xcode_14.0.0.app`

@IDispose that should not matter, quick question have you opened Xcode.app at least once since you upgraded? Does a File new iOS blank project works for you? If the issue persist please open a new bug report and fill in the provided template with your details and we can take it from there.

**Comment by IDispose** [comment]
@dalexsoto Yes. I restarted the machine, opened XCode once and VS 2022 couple times. Even a brand new project exhibits the same behavior.

**Comment by jburtjr** [comment]
Is there any timeline on when the .net Maui will be compatible with Xcode 14?  We have a product needing to be released but now I have no way of deploying the product to the App store.  I tried downgrading my Xcode to 13.4.1 but still haven't had any success getting the app to run in the simulators or creating a package to deploy to the app store.

**Comment by dalexsoto** [comment]
> Is there any timeline on when the .net Maui will be compatible with Xcode 14? We have a product needing to be released but now I have no way of deploying the product to the App store. I tried downgrading my Xcode to 13.4.1 but still haven't had any success getting the app to run in the simulators or creating a package to deploy to the app store.

Mmm downgrading Xcode should let you be in the state before Xcode 14, your issue seems different and somewhat unrelated to which Xcode you have on your machine, this is just a pure guess from the info provided. Please file an issue agains the MAUI repo https://github.com/dotnet/maui/issues and fill in the issue template and link back here or ping me and we can have a look.

About a timeline we should have manual steps to unblock you any .NET6/MAUI users soon too, I'll reply to this thread as son as we have them.

**Comment by ac2112** [comment]
James Montemago created a YouTube video on How to install Multiple versions of Xcode on your Mac. Once this issue is fixed, we're all going to have to deal with it again when Apple performs another update. This is a simple way to manage releases on our side. [](https://youtu.be/41h0xIKusT0)(url)

**Comment by l3bel** [comment]
Hello,
Thanks everyone for the huge efforts to fix these issues.
I confirm there are no more registrar build errors at this point [xamarin.ios-16.0.0.72.pkg](https://bosstoragemirror.blob.core.windows.net/wrench/6.0.4xx-xcode14/6756a11462d8aa6e73d4a320712b276caba159c1/6718459/package/notarized/xamarin.ios-16.0.0.72.pkg). I now have networking related issues with other parts (a nuget), but I think it's that nuget to blame.
Thanks again to everyone who contributed their time and efforts -and still do

**Comment by Magic73** [comment]
> I wrote an article guiding others with the possibilities(using the build from this issue, using Xcodes to backup / handling multiple XCode versions, wait for the update. Please ping me if you have released the update for full support, I'll check periodically myself, too. Updating the article when update has been shipped.
> 
> You can find the post here: https://eekayonline.medium.com/fixing-the-xamarin-ios-build-error-mt4109-failed-to-compile-the-generated-registrar-code-8856179427b8
> 
> When the💩 hits the fan, it's easy to forget the amount of work being put into keeping up with other crew's stuff. So, here's a shoutout to @dalexsoto and the maintainers of the repo for their quick interaction.
> 
> THANKS and looking forward to the update :)

I got tons of errors while installing cask xcodes :(

**Comment by EeKay** [comment]
> James Montemago created a YouTube video on How to install Multiple versions of Xcode on your Mac. Once this issue is fixed, we're all going to have to deal with it again when Apple performs another update. This is a simple way to manage releases on our side. https://youtu.be/oC5zpEbwViE

Thanks I'll mention this in my article too! (Also thanks to @Magic73 )

**Comment by Magic73** [comment]
Anyway, it doesn't work to me.  if I install two xcode packages, neither of them work. when I try to start one, it freezes and I have to close it forcibly. It doesn't matter which version I keep active. Xcode only starts working again if I uninstall one of the two versions.
And my problem is not solved, because the version of Visual Studio I have is now too updated compared to the version of xcode I should be using (13.3, with SDK 15.4).

**Comment by ivanicin** [comment]
I get this when submitting to the App Store:

> ITMS-90398: Invalid WatchKit Support - The bundle contains an invalid implementation of WatchKit. The app may have been built or signed with non-compliant or pre-release tools. Visit developer.apple.com for more information. 

At this point I don't see any reason from this. I did have a similar problem when I had XCode beta installed, but I don't have it anymore and it worked properly when I uninstalled XCode beta.

I'll experiment some more but this looks to be Xamarin problem.

**Comment by ivanicin** [comment]
One warring thing is that I see in the App Center huge pile of these coming from my test device:

```
Context.Finalize ()
SIGABRT: Linked Away
```

I don't think that those were crashing the app in use though they are indicated as crashes in AppCenter. Something like this has never happened before Xcode 14.

**Comment by dalexsoto** [comment]
From Xcode 14.0 [release notes](https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes):

> Deprecations
> 
> Starting with Xcode 14, bitcode is no longer required for watchOS and tvOS applications, and the App Store no longer accepts bitcode submissions from Xcode 14.
>
> Xcode no longer builds bitcode by default and generates a warning message if a project explicitly enables bitcode: “Building with bitcode is deprecated. Please update your project and/or target settings to disable bitcode.” The capability to build with bitcode will be removed in a future Xcode release. IPAs that contain bitcode will have the bitcode stripped before being submitted to the App Store. Debug symbols for past bitcode submissions remain available for download. (86118779)

@ivanicin this is related to Xcode 14 but can you please file a separate bug report and fill in the template for this? Looks like we will need to have a deeper look at this.

**Comment by harsh9592** [comment]
I tried xamarin.ios 16.0. 
No luck.
App worked fine on emulator iphone 14 ios 16.0
But on physical device iphone 12 ios 16.0. not able to install from enterprise(release mode).(yes my app is on enterprise release and not on app store).
Fyi not able to try debug on iphone 12 ios 16.
Any suggestions?

**Comment by RonnyRos** [comment]
No, having the same issues, physical device does not work.

**Comment by DamianMehers** [comment]
FWIW I'm running on real devices - it did give me an error but when I looked at the "Deploying to device" tab it showed an error re provisioning.  Once I'd fixed that (by creating an xCode project using the same bundle id, signing identity and provisioning profile and then letting xCode fix everything) it worked.

**Comment by RonnyRos** [comment]
Can confirm @DamianMehers comment works, recreate the project in xcode with the same bundleid, restart your VSMac, it should deploy to your ios 16 device.

**Comment by ivanicin** [comment]
@DamianMehers this has worked for me without touching anything, but I had to create new provisional profiles as I had some new capabilities. So it is likely that just creating new provisional profiles is enough if that is easier.

**Comment by harsh9592** [comment]
Failed.
I tried with a new provisioning profile and after release on iphone 12 ios16.0.2 via enterprise, it still shows the same error. 
'Unable to istall'

When I try to install from the enterprise app store, it starts downloading but after downloading and at time of install, it stops and shows a little cloud icon with a down(download) arrow before the app name, and when I tap on it, it shows an error as mentioned.
Steps i performed:
1. Created a new provisioning profile from an existing certificate.
2. Download and add to xcode.
3. Replaced in vs 2022 17.3.5(build 0).
4. Closed all and reopened.
5. Clean and rebuild multiple times, check profile settings again.
6. Created archive, published on enterprise store.
7. Download on iphone 12 ios16.0.2
Error.
Any steps i missed?
Any other solution?

**Comment by kumarc-123** [comment]
> I Installed the 16.0 package pinned at the top. I tryed to deploy to an attached device that was working before upgrade to XCode 14. I get the following error: error HE0004: Could not load the framework 'DVTITunesSoftwareServiceFoundation' (path: /Applications/Xcode.app/Contents/SharedFrameworks/DVTITunesSoftwareServiceFoundation.framework/DVTITunesSoftwareServiceFoundation). This problem is probably related to the comment of @ss-proc ([#15954 (comment)](https://github.com/xamarin/xamarin-macios/issues/15954#issuecomment-1257040075))
> 
> I'm also unable to see any simulator.
> 
> Tips are very welcome, thanks

Facing the same issue ehre

**Comment by elomonaco** [comment]
xamarin.ios-16.0.0.72 works like a charm for me, I can now run my xamarin.ios app while still having Xcode 14 installed 😄 

appreciate the work you guys put into getting this working, I do native ios development too, so worrying about juggling multiple xcode versions was going to be a hassle.

**Comment by awaescher** [comment]
The update is now available within VS for Mac. Big hugs to everybody that worked on that 🫂

<img width="845" alt="Bildschirmfoto 2022-09-27 um 20 28 33" src="https://user-images.githubusercontent.com/3630638/192626896-12388eac-ddda-4d2d-9007-3c94da53276a.png">

For me, selecting iOS simulators worked immediately after the update but I could not deploy onto them. 
The application output was:

```
xcrun: error: unable to find utility "simctl", not a developer tool or in PATH
error HE0046: Failed to install the app '...' on the device 'iOS 16.0 (20A360) - iPhone 14 Pro Max': simctl returned exit code 72
```

The fix was indeed opening Xcode and telling it where its own command line tools are located.
- [How do I fix the xcrun unable to find simctl error?](https://stackoverflow.com/questions/29108172/how-do-i-fix-the-xcrun-unable-to-find-simctl-error)](https://stackoverflow.com/a/63172300/704281)

**Comment by dalexsoto** [comment]
As you found out xamarin.ios-16.0.0.72 is now available in both Visual Studio for Mac and Visual Studio for Windows 17.3.5. Thanks a lot for your patience. I am closing and locking down this issue. The Xcode support for MAUI is still being worked out, I will open a separate issue to track this one.

Again thanks everyone for your patience!
